test/controllers/songs_controller_test.rb

```diff
@@ -6,21 +6,42 @@ class SongsControllerTest < ActionController::TestCase
     assert_response :success
   end
 
-  test "index should retrieve relevant search results" do
-    get :index, sSearch: "hand", format: :json, xhr: true
+  test "index should retrieve relevant keyword search results" do
+    get :index, search: { value: "hand" }, format: :json, xhr: true
 
-    songs_data = JSON.parse(@response.body)["data"].map! do|s|
-      s.delete('relevance')
-      Song.new(s)
-    end
+    songs_data = load_songs
 
     assert_includes(songs_data, songs(:hands_to_the_heaven))
     assert_includes(songs_data, songs(:glorious_day))
   end
 
+  test "index should stack searches" do
+    http_params = {
+      search: { value: "forever" },
+      tempo: "Medium",
+      key: "C",
+      format: :json,
+      xhr: :true
+    }
+
+    get :index, http_params
+    songs_data = load_songs
+
+    assert_includes(songs_data, songs(:all_my_hope))
+    assert_includes(songs_data, songs(:forever_reign))
```

Comment:
What I was saying for this test is you should assert that two songs with two of the three filters matching are NOT in the results. You are checking if songs that match all three filters are in it, but a empty search or one with just one of the filters would return all of them anyway, but also some more, so you want to check that almost matching songs aren't in it.